### PR TITLE
Fix MarkdownRenderer bugs

### PR DESCRIPTION
**Pull request by jaybhagat**
## Notion ticket link

[Fix Markdown Renderer issues](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=4876731ab8b0454f878f404087a4fcdd)

## Implementation description

* Underline: ReactMarkdown does not distinguish between `**` and `__`, so a custom component was created that distinguishes between the two.
* Firefox: Issue on Firefox was fixed (by adding styling)
* Text alignment: Text alignment is implemented for regular text and headers (including pictures, videos, etc.) - when ReactMarkdown recognizes a header or body text, custom styling is added to it

NOTE: text alignment for `h1` was not implemented because it messed with the highlighting (background). To align the text, `h1` cannot have `display: inline`, but it needs it or the background will mess up when there is more than one line.

## Steps to test

1. Go to http://localhost:3000/admin/dashboard/builder
2. Select Content Type as 'Text / Image / Video'
3. Use the underline button or `__` to wrap text you want to underline
4. Use `\\`, `||`, `//` wrappers to align the text
5. For headers, be sure to include the text alignment wrappers AFTER the header specification (i.e. `# \\ text \\`)

Screen Shot 2022-01-29 at 1 23 10 PM
Screen Shot 2022-01-29 at 12 17 18 PM

## What should reviewers focus on?

If the content is properly aligned and underlining works as intended. If anyone has any ideas to align `h1`, feel free to let me know.

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] I have run the appropriate linter(s)

### CODE REVIEW

**Review by chamod-gamage — APPROVED** [review]
Amazing work @jaybhagat ! This is huge for the module builder.
1 small ask: 
Do you mind going into markdownEditor/index.tsx, into each of the align commands, and making it so that the selection area upon alignment change includes the slashes and any internal \\, ||, // get replaced out with empty strings?
For example the rightAlignCommand may have this at the bottom: 
```
const state2 = api.replaceSelection(
            `//${state1.selectedText.replaceAll(/\\\\|\|\||\/\//g, "")}//`,
        ); // convention: right align is double //
        // Adjust the selection to not contain the **
        api.setSelectionRange({
            start: state2.selection.end - state1.selectedText.length + 4,
            end: state2.selection.end,
        });
```

File: src/components/markdownRenderer/index.tsx [hunk 1]
```diff
@@ -14,6 +15,35 @@ const stringConversionObj = {
     "<>[": "[",
 };
 
+// aligns the text accordingly for p + h tags (except h1)
```
Comment (ChinemeremChigbo) [inline src/components/markdownRenderer/index.tsx:18]:
```suggestion
// aligns the text accordingly for p + h tags
```

File: src/components/markdownRenderer/markdown-renderer.module.css [hunk 2]
```diff
@@ -7,6 +7,7 @@
         #7cf952 55%
     );
     text-decoration: underline;
+    text-decoration-skip-ink: none;
     text-decoration-thickness: 19px;
     text-decoration-color: #7cf952;
     display: inline;
```
Comment (ChinemeremChigbo) [inline src/components/markdownRenderer/markdown-renderer.module.css:13]: To fix the h1 issue you could do:
.markdown h1 {
    font-size: 48px;
    color: black;
    text-decoration: underline;
    text-decoration-skip-ink: none;
    text-decoration-thickness: 50px;
    text-underline-offset: -40px;
    text-decoration-color: #7cf952;
}

**Review by ChinemeremChigbo — CHANGES_REQUESTED** [review]
Great work! See recommendations.

**Review by tawnyzhao — APPROVED** [review]
Looks good!

File: src/components/markdownRenderer/markdown-renderer.module.css (hunk 2 above)
Comment (jaybhagat) [reply src/components/markdownRenderer/markdown-renderer.module.css]: @ChinemeremChigbo take a look when you're free! Thanks for the suggestion :)

**Review by ChinemeremChigbo — APPROVED** [review]
Nice!

**Pull request merged**